fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -800,7 +810,80 @@ public static FHIRSearchContext parseQueryParameters(Class<?> resourceType,
         context.setSearchParameters(parameters);
         return context;
     }
+    
+    /**
+     * Checks the query parameters (with the same parameter code) against any search restrictions specified
+     * in the SearchParameter resource for that parameter code.
+     * 
+     * @param parameterCode
+     *            the parameter code
+     * @param searchParameter
+     *            the SearchParameter resource
+     * @param queryParameters
+     *            the query parameters to check
+     * @throws FHIRSearchException
+     *             if a search restriction is found that is not followed
+     */
+    private static void checkSearchParameterRestrictions(String parameterCode, SearchParameter searchParameter, List<QueryParameter> queryParameters)
+        throws FHIRSearchException {
+
```

Comment:
I see all the failures.  We may want to add a configuration to turn this feature on/off.

Reply:
Hopefully not. I was using the Draft Pull Request to run the e2e tests before this is ready for review, and noticed a fix I needed to make.  We'll see if that takes care of the errors.